### PR TITLE
Clean up SharedAccessSignature files for the hub service so that each class has a distinct responsibility

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
On the hub service side, separate out the constants, parsers, builder and model for shared access signature, like what we have on the hub device side.

### CODE REVIEW

File: iothub/service/src/Authentication/Security/SharedAccessSignature.cs [hunk 1]
```diff
@@ -63,79 +61,6 @@ internal sealed class SharedAccessSignature : ISharedAccessSignatureCredential
         /// </summary>
         public string Signature { get; private set; }
```
Comment (abhipsaMisra) [inline iothub/service/src/Authentication/Security/SharedAccessSignature.cs:62]: you can make these properties internal as well

File: iothub/service/src/Authentication/Security/SharedAccessSignatureParser.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+{
+    internal sealed class SharedAccessSignatureParser
+    {
+        /// <summary>
+        /// Parses a shared access signature string representation into a <see cref="SharedAccessSignature"/>.
+        /// </summary>
+        /// <param name="iotHubName">The IoT hub name.</param>
+        /// <param name="rawToken">The string representation of the SAS token to parse.</param>
+        /// <returns>The <see cref="SharedAccessSignature"/> instance that represents the passed in raw token.</returns>
+        public static SharedAccessSignature Parse(string iotHubName, string rawToken)
```
Comment (abhipsaMisra) [inline iothub/service/src/Authentication/Security/SharedAccessSignatureParser.cs:19]: this too, can be internal

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/service/src/Authentication/Security/SharedAccessSignatureParser.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+namespace Microsoft.Azure.Devices
+{
+    internal sealed class SharedAccessSignatureParser
+    {
+        /// <summary>
+        /// Parses a shared access signature string representation into a <see cref="SharedAccessSignature"/>.
+        /// </summary>
+        /// <param name="iotHubName">The IoT hub name.</param>
+        /// <param name="rawToken">The string representation of the SAS token to parse.</param>
+        /// <returns>The <see cref="SharedAccessSignature"/> instance that represents the passed in raw token.</returns>
```
Comment (drwill-ms) [inline iothub/service/src/Authentication/Security/SharedAccessSignatureParser.cs:18]: <see> tag not needed. The type is already evident and linkable by being part of the API definition.
```suggestion
        /// <returns>The shared access signature instance that represents the passed in raw token.</returns>
```

File: iothub/service/src/Authentication/Security/SharedAccessSignatureParser.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        public static SharedAccessSignature Parse(string iotHubName, string rawToken)
+        {
+            if (string.IsNullOrWhiteSpace(iotHubName))
```
Comment (drwill-ms) [inline iothub/service/src/Authentication/Security/SharedAccessSignatureParser.cs:21]: Argument.AssertNotNullOrWhiteSpace?

Also below.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**